src/libraries/System.Text.Json/tests/JsonNode/JsonNodeOperatorTests.cs

```diff
@@ -0,0 +1,219 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Xunit;
+
+namespace System.Text.Json.Node.Tests
+{
+    public static class OperatorTests
+    {
+        private const string ExpectedPrimitiveJson =
+                @"{" +
+                @"""MyInt16"":1," +
+                @"""MyInt32"":2," +
+                @"""MyInt64"":3," +
+                @"""MyUInt16"":4," +
+                @"""MyUInt32"":5," +
+                @"""MyUInt64"":6," +
+                @"""MyByte"":7," +
+                @"""MySByte"":8," +
+                @"""MyChar"":""a""," +
+                @"""MyString"":""Hello""," +
+                @"""MyBooleanTrue"":true," +
+                @"""MyBooleanFalse"":false," +
+                @"""MySingle"":1.1," +
+                @"""MyDouble"":2.2," +
+                @"""MyDecimal"":3.3," +
+                @"""MyDateTime"":""2019-01-30T12:01:02Z""," +
+                @"""MyDateTimeOffset"":""2019-01-30T12:01:02+01:00""," +
+                @"""MyGuid"":""1b33498a-7b7d-4dda-9c13-f6aa4ab449a6""" + // note lowercase
+                @"}";
+
+        [Fact]
+        public static void ImplicitOperators_FromProperties()
+        {
+            var jObject = new JsonObject();
+            jObject["MyInt16"] = (short)1;
+            jObject["MyInt32"] = 2;
+            jObject["MyInt64"] = (long)3;
+            jObject["MyUInt16"] = (ushort)4;
+            jObject["MyUInt32"] = (uint)5;
+            jObject["MyUInt64"] = (ulong)6;
+            jObject["MyByte"] = (byte)7;
+            jObject["MySByte"] = (sbyte)8;
+            jObject["MyChar"] = 'a';
+            jObject["MyString"] = "Hello";
+            jObject["MyBooleanTrue"] = true;
+            jObject["MyBooleanFalse"] = false;
+            jObject["MySingle"] = 1.1f;
+            jObject["MyDouble"] = 2.2d;
+            jObject["MyDecimal"] = 3.3m;
+            jObject["MyDateTime"] = new DateTime(2019, 1, 30, 12, 1, 2, DateTimeKind.Utc);
+            jObject["MyDateTimeOffset"] = new DateTimeOffset(2019, 1, 30, 12, 1, 2, new TimeSpan(1, 0, 0));
+            jObject["MyGuid"] = new Guid("1B33498A-7B7D-4DDA-9C13-F6AA4AB449A6");
+
+            string expected = ExpectedPrimitiveJson;
+
+            string json = jObject.ToJsonString();
+
+            // Adjust for non-Core frameworks which do not have round-trippable floating point strings.
+            json = json.Replace("1.10000002", "1.1").Replace("2.2000000000000002", "2.2");
+
+            Assert.Equal(expected, json);
```

Comment:
Is the serialization order guaranteed here (pending an ordered-dictionary-based implementation)?

Reply:
Yes pending the ordered dictionary which will be added after this PR. Currently the test workswith `Dictionary<,>`, although that technically is not deterministic and only works for add-only cases (not items removed).